### PR TITLE
Auto

### PR DESCRIPTION
**Pull request by NoahRoseLedesma**
PR for reviews

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        RelicRecoveryVuMark mark = vHelper.getVuMark();
+        telemetry.log().add("DETECTED COLUMN: " + mark);
+
+        // Kick the jewel off.
+        doJewel();
+
+        if (startPos == StartPosition.BLUE_A || startPos == StartPosition.RED_A) {
+            switch (mark) {
+                case LEFT:
+                    m.run(1900, 0, 1);
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: For these run commands consider resolving the magic numbers.
There are quite a few of these so perhaps you could create a data structure to represent the different run commands eg: `MyRobotMovements::RED_A_START_TO_CENTER = [600, 0, 1];` then unpack the array in the m.run call

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+                    break;
+                case RIGHT:
+                    m.setTweenTime(0);
+                    m.run(600, 0, 1);
+                    m.setTweenTime(TWEEN_TIME);
+                    break;
+            }
+        } else {
+            telemetry.log().add("Executing on position B");
+            moveVec.SetComponents(0, 1);
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: This is on me but I should create static final unit vectors eg: UNIT_VECTOR_IHAT or UNIT_VECTOR_JHAT

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            m.setDirectionVector(moveVec);
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: Relating to the previous comment. In which case we could say `m.setDirectionVector(UNIT_VECTOR_IHAT)`

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                        m.run(1750, 0, 1);
+                        telemetry.log().add("Finished running, starting turn");
+                        m.setRotationTarget(-90*sideModifier);
+                        m.turnToTarget();
+                    } else {
+                        telemetry.log().add("CASE LEFT");
+                        m.run(2400, 0, 1);
+                        m.setRotationTarget(-90 * sideModifier);
+                        m.turnToTarget();
+                        moveVec.SetComponents(-1 * sideModifier, 0);
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: Want me to create vector transformation functions?

Comment (NoahRoseLedesma) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: Flip vertically, flip horizontally, etc

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+            telemetry.log().add("Executing on position B");
+            moveVec.SetComponents(0, 1);
+            m.setDirectionVector(moveVec);
+            switch (mark) {
+                case LEFT:
+                    if (startPos == StartPosition.BLUE_A || startPos == StartPosition.BLUE_B) {
+                        telemetry.log().add("CASE RIGHT");
+                        m.run(1750, 0, 1);
+                        telemetry.log().add("Finished running, starting turn");
+                        m.setRotationTarget(-90*sideModifier);
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: `-90*sideModifier` -> `-90 * sideModifier`

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+                        m.setTweenTime(0);
+                        m.run(600, 0, 1);
+                        m.setTweenTime(TWEEN_TIME);
+                    }
+                    break;
+                case CENTER:
+                    telemetry.log().add("CASE CENTER");
+                    m.run(2400, 0, 1);
+                    telemetry.log().add("Finished running, starting turn");
+                    m.setRotationTarget(-90*sideModifier);
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: `-90*sideModifier` -> `-90 * sideModifier`

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java (hunk 1 above)
Comment (NoahRoseLedesma) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: Did some reading, unpacking is hard but you can cheat: Overload m.run to accept an array of three ints :neckbeard:

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+            moveVec.SetComponents(0, 1);
+            m.setDirectionVector(moveVec);
+            m.run(2200, 0, 1);
+        } else {
+            m.setTweenTime(0);
+            moveVec.SetComponents(0, 1);
+            m.setDirectionVector(moveVec);
+            m.run(300, 0, 1);
+            m.setTweenTime(TWEEN_TIME);
+        }
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: This whole block of code should be one function. navToCryptoColumn()?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+                hardwareMap.dcMotor.get("m3"),
+                hardwareMap.get(BNO055IMU.class, "imu"),
+                this
+        );
+        m.debugModeEnabled = true;
+        // Set global tween time.
+        m.setTweenTime(TWEEN_TIME);
+
+        // Strafe off motion setup.
+        moveVec = new Vector2D(-1*sideModifier, 0);
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java:81]: `1*sideModifier` -> `1 * sideModifier`

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+        rTentacle.setPosition(ServoValue.RIGHT_TENTACLE_UP);
+        lTentacle.setPosition(ServoValue.LEFT_TENTACLE_UP);
+
+        jsL = hardwareMap.colorSensor.get("jsLeft");
+        jsR = hardwareMap.colorSensor.get("jsRight");
+    }
+
+    void depositBlock() {
+        lift.setPower(1);
+        sleep(1000);
```
Comment (NoahRoseLedesma) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: Some of these random sleeps you can probably leave as magic numbers

**Review by NoahRoseLedesma — COMMENTED** [review]
:1st_place_medal:

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java (hunk 1 above)
Comment (mattkelsey) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: I am viewing the actual OpMode as the place for magic numbers sort of like a movement configuration. This seems like adding an extra abstraction layer on top that is not necessary... Idk could be wrong

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java (hunk 4 above)
Comment (mattkelsey) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TesseractAuto.java]: That'd be great

**Comment by mattkelsey** [comment]
Closing, manually merging, suggested changes finished :)

**Pull request closed without merging**